src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +691,433 @@ protected function limit_list( $key, $list ) {
+		}
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data, $ticket );
+
+		// Update the attendee data for referencing with what we handled in the set_attendee_args().
+		$attendee_data = array_merge( $attendee_data, $this->updates );
+
+		// Create the new attendee.
+		$attendee = $this->create();
+
+		// Handle any further attendee updates.
+		$this->save_extra_attendee_data( $attendee, $attendee_data, $ticket );
+
+		// Trigger creation actions.
+		$this->trigger_create_actions( $attendee, $attendee_data, $ticket );
+
+		// Clear the attendee cache if post_id is provided.
+		if ( ! empty( $this->updates['post_id'] ) ) {
+			$this->attendee_provider->clear_attendees_cache( $this->updates['post_id'] );
+		}
+
+		return $attendee;
+	}
+
+	/**
+	 * Create an attendee object from ticket and attendee data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $attendee_data  List of attendee data to be saved.
+	 * @param bool  $return_promise Whether to return a promise object or just the ids
+	 *                              of the updated posts; if `true` then a promise will
+	 *                              be returned whether the update is happening in background
+	 *                              or not.
+	 *
+	 * @return array|Tribe__Promise A list of the post IDs that have been (synchronous) or will
+	 *                              be (asynchronous) updated if `$return_promise` is set to `false`;
+	 *                              the Promise object if `$return_promise` is set to `true`.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function update_attendee( $attendee_data, $return_promise = false ) {
+		if ( empty( $attendee_data['attendee_id'] ) ) {
+			throw new Tribe__Repository__Usage_Error( 'You must provide the attendee_id when updating an attendee from the Attendees Repository class' );
+		}
+
+		$this->by( 'id', $attendee_data['attendee_id'] );
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data );
+
+		// Update the attendee.
+		$saved = $this->save( $return_promise );
+
+		if ( $return_promise ) {
+			$repository = $this;
+
+			return $saved->then( static function() use ( $repository, $attendee_data ) {
+				// Trigger the update actions.
+				$repository->trigger_update_actions( $attendee_data );
+			} );
+		}
+
+		// Trigger the update actions.
+		$this->trigger_update_actions( $attendee_data );
+
+		return $saved;
+	}
+
+	/**
+	 * Set arguments for attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function set_attendee_args( $attendee_data, $ticket = null ) {
```

Comment:
:warning: Function's nesting level (6) exceeds 5; consider refactoring the function

<sub>hash: 8329d9b1f7a525e2b8b28b6f7f864e1a</sub>